pkg/server/test/write_authzmodel.go

```diff
@@ -135,6 +136,21 @@ func WriteAuthorizationModelTest(t *testing.T, datastore storage.OpenFGADatastor
 				},
 			},
 		},
+		{
+			name: "self_referencing_type_restriction_with_entrypoint",
+			request: &openfgapb.WriteAuthorizationModelRequest{
+				StoreId: storeID,
+				TypeDefinitions: parser.MustParse(`
```

Comment:
Can we add a test for a case like this (it must fail validation):
```py
model
  schema 1.1
type user
type document
  relations
    define editor: [user] as self
    define viewer: [document#viewer] and editor
```

Reply:
This test case introduces some significant complexities that will need to be addressed. I'm going to create an issue to circle back to this test case and makes sure it gets addressed.

Reply:
Good test! It exposed a little bug with the logic. I've fixed it and added this test as well as a similar one for a model involving exclusion (e.g `but not`).